languages.toml

```
@@ -695,7 +695,7 @@ roots = ["pyproject.toml", "setup.py", "poetry.lock", "pyrightconfig.json"]
 comment-token = "#"
 language-servers = [ "pylsp" ]
 # TODO: pyls needs utf-8 offsets
-indent = { tab-width = 4, unit = "    " }
+indent = { tab-width = 4, unit = "    ", required = true }
```

```suggestion
indent = { tab-width = 4, unit = "    " }
```
Python allows using hard tabs for indentation (and actually any amount of spaces works too), it only disallows ambiguous mixing (i.e. when semantics depend on a specific tab width)

---

I didn't mark the indentation style as `required` because Python requires it. I did it in order to override the global setting for conformance with PEP 8.

---

I think `required` should only be applied to languages that do not support using tabs (like Nim) or spaces (like Makefile); everything else is basically just your stylistic choice which shouldn't affect anyone else. I mean, that's exactly the reason for using tabs instead of spaces.

---

I agree—I use tabs in Python and everywhere else—but this felt right to me at the time. In any case, I'll leave the decision to the maintainers.